### PR TITLE
refactor(frontend): merge steps for resolving on-conflict behavior in create table handler

### PR DESCRIPTION
**Pull request by BugenZhao**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://raw.githubusercontent.com/risingwavelabs/risingwave/17af8a747593ebdbfa826691daf75bdab7d14fa0/.github/contributor-license-agreement.txt).

## What's changed and what's your intention?

Merge the steps for resolving on-conflict behavior in `CREATE TABLE` handler.

Also support passing an resolved protobuf repr for on-conflict behavior. This is a preparation for #19866.

## Checklist

- [x] I have written necessary rustdoc comments.
- [ ]  I have added necessary unit tests and integration tests.
- [ ]  I have added test labels as necessary. 
- [ ]  I have added fuzzing tests or opened an issue to track them. 
- [ ]  My PR contains breaking changes. 
- [ ]  My PR changes performance-critical code, so I will run (micro) benchmarks and present the results. 
- [ ]  My PR contains critical fixes that are necessary to be merged into the latest release. 

## Documentation

- [ ]  My PR needs documentation updates. 

<details>
<summary><b>Release note</b></summary>

</details>

### CODE REVIEW

**Comment by BugenZhao** [comment]
* **#19917** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19917?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#19916** <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19916?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/risingwavelabs/risingwave/19916?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: src/frontend/src/handler/create_table.rs [hunk 1]
```diff
@@ -667,6 +668,25 @@ fn gen_table_plan_with_source(
...
+#[derive(Clone, Copy)]
+pub enum EitherOnConflict {
+    Ast(Option<OnConflict>),
+    Resolved(ConflictBehavior),
+}
+
+impl From<Option<OnConflict>> for EitherOnConflict {
+    fn from(v: Option<OnConflict>) -> Self {
+        Self::Ast(v)
+    }
```
Comment (st1page) [inline src/frontend/src/handler/create_table.rs:681]: Why not just transform the Ast struct to `ConflictBehavior` here?

File: src/frontend/src/optimizer/mod.rs [hunk 2]
```diff
@@ -849,15 +849,35 @@ impl PlanRoot {
...
+                } else {
+                    // Default to `UPDATE FULL` for non-append-only tables.
+                    match on_conflict.unwrap_or(OnConflict::UpdateFull) {
+                        OnConflict::UpdateFull => ConflictBehavior::Overwrite,
+                        OnConflict::Nothing => ConflictBehavior::IgnoreConflict,
+                        OnConflict::UpdateIfNotNull => ConflictBehavior::DoUpdateIfNotNull,
+                    }
+                }
+            }
+            EitherOnConflict::Resolved(b) => b,
```
Comment (xxchan) [inline src/frontend/src/optimizer/mod.rs:880]: What about making this a method on `EitherOnConflict` to make the optimizer's code shorter?

**Review by xxchan — APPROVED** [review]
😁

**Pull request merged**